### DISCUSSION TITLE
Battery NOT charging from AC with full (5KW) power

### DISCUSSION THREAD

**Discussion by AMichaelSuess** [WillCodeForCats/solaredge-modbus-multi#469] [General]
Dear all, also trying my luck here, as maybe somebody with a home battery can help. When I change the `Storage Command Mode` to `Charge from Solar Power and Grid`, my battery charges with less than 1KW. It usually is able to charge with 5KW - and I have seen it doing that on solar power. I have the `Storage Charge Limit` set to 5000W - and changing that also while the operation is running does nothing to increase the power. I have `AC Charge Policy` set to `Always Allowed` - therefore also no `AC Charge Limit` is set - or should be set according to the documentation.

So what am I missing? How do I get my battery to charge with full 5KW power? If anybody in this round has any ideas - I would appreciate any pointers! Thanks!

**Comment by darkwind0** [comment]
Fairly certain when charging from AC the maximum is 1KW.  My SE7600H-USSNBBL14 will only grid charge the same at 1KW as you have observed.  I can get up to 5KW from solar as well.

I am not able to find a relevant spec sheet that shows this value, however hopefully the knowledge is helpful.

-darkwind

**Comment by AMichaelSuess** [comment]
Hmm, there appear to be a couple of people posting in the HA-forum where things appear to work up to 5KW - such as [here ](https://community.home-assistant.io/t/solaredge-modbus-configuration-for-single-inverter-and-battery/464084/138?u=misu2410)or [here](https://community.home-assistant.io/t/solaredge-modbus-configuration-for-single-inverter-and-battery/464084/122?u=misu2410). I can of course not rule out that they have a different inverter from mine.

**Comment by AMichaelSuess** [comment]
OK, found my problem. I had not set `Reactive Power mode to RRCR` - as is advised at the bottom of [this page](https://github.com/binsentsu/home-assistant-solaredge-modbus). I had only set `Set Advanced Power Control to Enable`. With this setting applied, my charge-rate went up to 5KW. Thanks for the help!

**Reply by WillCodeForCats** [reply]
Thanks for the config tip; I'll get that added somewhere.

**Reply by WillCodeForCats** [reply]
This setting will be added in [PR #517 Add Reactive Power Mode select entity](https://github.com/WillCodeForCats/solaredge-modbus-multi/pull/517)

**Reply by darkwind0** [reply]
Thee same settings have no effect for me, my inverter grid-charges at 1kw no matter the power control settings. however Iam apparently one of the 'unlucky ones' who do not have power control options nor any battery options or viewing available to me no matter what I do. While I'd like to see it get to 5kw, realistically the only time I grid charge is impending storm alerts or a manual adjustment via SolarEdge's app.

I had lodged a case with SolarEdge and all they would do is direct me back to the cloud API.  I think getting the inverter adjusted to allow these functions is a hidden item not in their troubleshooting docs.

-darkwind

**Reply by AMichaelSuess** [reply]
Did you try using the installer interface from the inverter directly - as described in the page I have [linked above](https://github.com/binsentsu/home-assistant-solaredge-modbus)?

**Reply by darkwind0** [reply]
Yes, I did both steps -- Reactive power mode was changed (from CosPhi) to RRCR and I also enabled the RRCR interface with SetApp -- upon setting the battery to charge (by setting the max consumption reserve to 100%) it still only draws at max 1kw.  I verified this externally with CTs around the inverter AC connections that 500w is flowing on each phase.

**Reply by cyberkryten** [reply]
I also noticed this when changing the Reserve - it only charged at 2kW (I had two batteries at the time).  Nothing I did using the % Reserve value and configs charged any faster than 1kW per battery.

I do have Power Control options, and setting it to 'Charge from Solar and Grid' allowed charging at whatever the Charge Limit was set to.

So, with two batteries connected and the default 7kW (on my 8kW inverter) I got 3.5kW per battery.  

With three batteries and changing Charge Limit to 8kW I got 2.66kW per battery.

However, if I stopped that charge and then reset to Backup Only and increased the min charge level, it would still revert to only charging at 1kW per battery, so I think there's a limit somewhere, or a firmware issue (of which I am suffering a lot at present)

**Comment by deqw** [comment]
@AMichaelSuess Do you maybe have a setting or combination of settings to only use grid power to charge the battery? I mean disabling input from solar power and only charge the battery by grid. When I set `Active Power Limit` to 0 (zero) AC current on the inverter side is going to zero, but the battery remains DC charging from solar. I try to prevent charging the battery at some moments when tariff is below zero.

**Reply by AMichaelSuess** [reply]
Sorry, I have no clue there, as I am only charging from grid at night - and only very seldomly. My power prices also never go below 0...

**Comment by darkwind0** [comment]
I know this is an older topic, however I stumbled across something interesting -

If (in the Solaredge user app) the Battery mode is set to 'manual' then I can charge a single battery from grid at 5kw.  No scheduling or anything is required to be set, although it works with them as well.  Any other mode
used limits charge to 1kw rate.

I haven't tried changing the settings directly from the local inverter Wi-Fi but I suspect it would work there too.

Since my hardware is among those that do not expose the battery data/settings via modbus this is limited for me, but it answers the long-standing question hopefully!

-darkwind

**Comment by apf-doit** [comment]
my routine for grid charging above the limit is about the same:

`Advanced Power Control  → ON`
`AC Charge Policy        → Always Allowed`
`Limit Control Mode      → Disabled`
`Storage Control Mode    → Remote`
`Storage Command Mode    → Charge from Solar Power and Grid`
`Storage Charge Limit    → 5000`
`Storage Command Timeout → 14400`

(i have 18.4kWh to fill minus 10% backup)

i never had to use:

`Reactive Power Mode     → RRCR`

after all is done everything defaults back to normal because of that timeout is reached. without increasing the timeout for 5kW charging it stops exactly after 3600s and falls back to 1kW.

what you saying is to set the mode in the app: i looked at it but i need to set a schedule for that which does not make really sense to me, i want to charge when the price is low and not when a static schedule starts. but if that works it might be another option to overrule the limit.

**Comment by fredlcore** [comment]
Just FYI: There seems to be a bug/feature that requires to regularly set "Active Power Limit" (ModBus register 61441) to 100(%). While the manual says that the register is not stored during a reset of the inverter, it seems that mine needed a refresh every half day or so. Only then I could control the charge exactly based on "Remote Control Charge Limit" (ModBus register 57358). Hope that helps!
